src/workers/newNotificationMail.ts

```diff
@@ -501,8 +504,59 @@ const notificationToTemplateData: Record<NotificationType, TemplateDataFunc> = {
       ),
     };
   },
-  collection_updated: async () => {
-    return null;
+  collection_updated: async (con, user, notification) => {
+    const post = await con.getRepository(CollectionPost).findOne({
+      where: {
+        id: notification.referenceId,
+      },
+    });
+
+    if (!post) {
+      return;
+    }
+
+    const latestPostRelation = await con.getRepository(PostRelation).findOne({
+      where: {
+        postId: post.id,
+        type: PostRelationType.Collection,
+      },
+      order: {
+        createdAt: 'DESC',
+      },
+    });
+
+    if (!latestPostRelation) {
+      return;
+    }
+
+    const latestRelatedPost = await con.getRepository(ArticlePost).findOne({
+      where: {
+        id: latestPostRelation.relatedPostId,
+      },
+      relations: ['source'],
+    });
+
+    if (!latestRelatedPost) {
+      return;
+    }
```

Comment:
Doing 3 SQL queries seems wasteful to me. It also seems like we now need to handle the case of not having any data 3 times. Is it possible to do it in only 1 query? Or are we worried about the query complexity and future readability?

Reply:
Valid point, since this is done in the worker I don't think this is too impactful. I could optimize to two queries with second done in join but personally prefer readability in this case. Let's see what others think and I'll adjust.

Reply:
On the second thought, this does not look too bad, using `relations` to do the join on orm side which saves us one query. 6caf7e96cbe7b89b0ca5db5eb6ff84dd108c417f